### PR TITLE
feat(usage-report): Add decide usage report

### PR DESCRIPTION
**Pull request by neilkakkar**
## Problem

We want decide request analytics data in the usage report. This adds it.

A few considerations:

1. Data for all teams is sent to team 1 on EU, and team 2 on US.
2. We want to defend against a random bad actor sending us events to increase/decrease their usage. So, we only count 'valid' usage: where validity is controlled by a token we set (so only data coming from our own posthog prod instance, not even dev, is valid for usage counting)

In a follow up, I'll add the same for local evaluation requests.

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

Unit test

### CODE REVIEW

File: posthog/tasks/usage_report.py [hunk 1]
```diff
@@ -489,6 +512,10 @@ def send_all_org_usage_reports(
         # teams_with_event_count_by_name=get_teams_with_event_count_by_name(period_start, period_end),
         teams_with_recording_count_in_period=get_teams_with_recording_count_in_period(period_start, period_end),
         teams_with_recording_count_total=get_teams_with_recording_count_total(),
+        teams_with_decide_requests_in_period=get_teams_with_decide_requests_count_in_period(period_start, period_end),
```
Comment (neilkakkar) [inline posthog/tasks/usage_report.py:515]: Q: @raquelmsmith do you need total counts as well?

File: posthog/tasks/usage_report.py [hunk 2]
```diff
@@ -626,6 +653,12 @@ def send_all_org_usage_reports(
             ),
             recording_count_total=find_count_for_team_in_rows(team.id, all_data["teams_with_recording_count_total"]),
             group_types_total=find_count_for_team_in_rows(team.id, all_data["teams_with_group_types_total"]),
+            decide_requests_in_period=find_count_for_team_in_rows(
+                team.id, all_data["teams_with_decide_requests_in_period"]
```
Comment (raquelmsmith) [inline posthog/tasks/usage_report.py:657]:
```suggestion
                team.id, all_data["teams_with_decide_requests_count_in_period"]
```

File: posthog/tasks/usage_report.py [hunk 3]
```diff
@@ -626,6 +653,12 @@ def send_all_org_usage_reports(
...
+            ),
+            decide_requests_in_month=find_count_for_team_in_rows(
+                team.id, all_data["teams_with_decide_requests_in_month"]
```
Comment (raquelmsmith) [inline posthog/tasks/usage_report.py:660]:
```suggestion
            decide_requests_count_in_month=find_count_for_team_in_rows(
                team.id, all_data["teams_with_decide_requests_count_in_month"]
```

**Review by raquelmsmith — CHANGES_REQUESTED** [review]
Looks good! Just one nit for naming things consistently.

File: posthog/tasks/usage_report.py (hunk 1 above)
Comment (raquelmsmith) [reply posthog/tasks/usage_report.py:515]: Oh, people _may_ acutally use the `count_total` values in insights somewhere... but no need to add it here for now, we can always add later if we find that we want it for some reason.

File: posthog/tasks/test/test_usage_report.py [hunk 4]
```diff
@@ -543,6 +554,127 @@ def test_usage_report_hogql_queries(self) -> None:
...
+        _create_event(
+            distinct_id="3",
+            event="decide usage",
+            properties={"count": 20000, "token": "correct"},
+            timestamp=now() - relativedelta(days=20),
+            team=self.analytics_team,
+        )
+        flush_persons_and_events()
+
+        with self.settings(DECIDE_BILLING_ANALYTICS_TOKEN="correct"):
```
[the remark this answers is not in the document]
Comment (raquelmsmith) [reply posthog/tasks/test/test_usage_report.py:624]: Is this a question for me? No idea 😂 Write a test for it?

**Review by raquelmsmith — APPROVED** [review]

File: posthog/tasks/test/test_usage_report.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (neilkakkar) [reply posthog/tasks/test/test_usage_report.py:624]: No, for me, and yes, it's sorted 😋 . I'll tag you if it's for you 😂

**Pull request merged**